### PR TITLE
Prepare README for GA

### PR DESCRIPTION
**Pull request by jhaaaa**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -102,11 +102,6 @@ You can configure the client's network connection and key storage method with th
...
-| apiUrl         | Undefined             | Manually specify an API URL to use. If specified, value of `env` will be ignored.                   |
-| keyStoreType   | `networkTopicStoreV1` | Persist the wallet's key bundle to the network, or use `static` to provide private keys manually.   |
-| codecs         | `[TextCodec]`         | Add codecs to support additional content types. To learn more, see [Different types of content](#different-types-of-content). |
-| maxContentSize | `100M`                | Maximum message content size in bytes.                                                              |
-| appVersion     | Undefined             | Add a client app version identifier that's included with API requests. For example, you can use the following format: `appVersion: APP_NAME + '/' + APP_VERSION`. Setting this value provides telemetry that shows which apps are using the XMTP client SDK. This information can help XMTP developers provide app support, especially around communicating important SDK updates, including deprecations and required upgrades.      |
```
Comment (nakajima) [inline README.md]: These features still aren't in this SDK. I've been waiting to hear if anyone wants them before implementing them.

Comment (jhaaaa) [reply README.md]: Ah, thank you for the clarification, @nakajima - makes sense!

File: README.md [hunk 2]
```diff
@@ -102,17 +97,28 @@ let client = try Client.from(bundle: keys, options: .init(api: .init(env: .produ
...
+## Configure content types
+
+You can use custom content types by calling `Client.register`. The SDK comes with two commonly used content type codecs, `AttachmentCodec` and `RemoteAttachmentCodec`:
+
+```swift
+Client.register(AttachmentCodec())
+Client.register(RemoteAttachmentCodec())
+```
+
+To learn more about using `AttachmentCodec` and `RemoteAttachmentCodec`, see [Handle different content types](#handle-different-content-types).
```
Comment (jhaaaa) [inline README.md:121]: @nakajima I added this link to take folx to the deeper discussion on how to use these codecs. Does that make sense? <3

File: README.md [hunk 3]
```diff
@@ -290,21 +297,152 @@ let decodedConversation = containerAgain.decode(with: client)
 try await decodedConversation.send(text: "hi")
 ```
 
-### Different types of content
+## Handle different content types
+
+All of the send functions support `SendOptions` as an optional parameter. The `contentType` option allows specifying different types of content other than the default simple string standard content type, which is identified with content type identifier `ContentTypeText`. 
+
+To learn more about content types, see [Content types with XMTP](https://xmtp.org/docs/dev-concepts/content-types).
```
Comment (jhaaaa) [inline README.md:304]: Added this link as I think it might be helpful for folx to understand the basic concept of content types.

**Review by nakajima — APPROVED** [review]

**Pull request merged**